dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/shared/ImmutableEvent.java

```diff
@@ -41,7 +41,7 @@
 /**
  * @author Luciano Fiandesio
  */
-public class ImmutableEvent
+public class ImmutableEvent extends Event
```

Comment:
That is strange. Immutable event is a wrapper for event.
I don't think it should extend it

Reply:
why not ? and Immutable event is still an event. What is missing here is that all setters should throw an exception.
However it is a kind of "hack" because into the validators we have an Immutable event parameter while in pre hook we have an event, but I shared some logic between them
Actually I could add a "getEvent()" into ImmutableEvent, but that would really make it loose all truly immutable behaviours :)
Let's talk about how to do it without duplicating code

Reply:
Ok refactored the code a bit to avoid this hack. Thank you for spotting